### PR TITLE
docs: new article for demonstrating zot workflow with artifacts

### PR DESCRIPTION
**Pull request by mbshields**
documentation

**Which issue does this PR fix**:

none

**What does this PR do / Why do we need it**:

Demonstrates an end-to-end workflow for installing a zot registry, pushing and signing an image and artifact, and verifying the results.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/articles/workflow.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
+# Software Provenance Workflow Using OCI Artifact
```
Comment (rchincha) [inline docs/articles/workflow.md:1]: "Artifacts"

File: docs/articles/workflow.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+> :point_right: This article demonstrates an end-to-end workflow for installing a zot registry, pushing and signing an image and artifact, and verifying the results.
```
Comment (rchincha) [inline docs/articles/workflow.md:3]: "pushing an image and related artifact (such as an SBOM)"

Drop the ", and verifying..."

File: docs/articles/workflow.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+
+## Workflow
+
+### Step 1: Set up the environment and tools
```
Comment (rchincha) [inline docs/articles/workflow.md:8]: Pattern here is maybe ...

For the full script, go here etc 

But here is the script broken down with documentation/explanation.

**Comment by mbshields** [comment]
Questions:
What is the purpose of creating and truncating REF0?

How is REF0 used in the cosign commands in Step 6?

Should we show an example of the artifact tree?

File: docs/articles/workflow.md (hunk 1 above)
Comment (mbshields) [reply docs/articles/workflow.md]: Done

File: docs/articles/workflow.md (hunk 2 above)
Comment (mbshields) [reply docs/articles/workflow.md]: Done

File: docs/articles/workflow.md (hunk 3 above)
Comment (mbshields) [reply docs/articles/workflow.md:9]: Done

File: docs/articles/workflow.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+> :point_right: This article demonstrates an end-to-end workflow for installing a zot registry, then pushing and signing an image and a related artifact, such as an SBOM.
+
+## Workflow
+
+The following sections describe the step-by-step workflow. To view the steps combined into a single script, see [Reference: Full workflow script](#fullscript).
+
+### Step 1: Set up the environment and tools
+
+As a first step, we define and create directories, then download binaries for zot, regctl, and cosign.
```
Comment (rchincha) [inline docs/articles/workflow.md:11]: ## Download client tools

File: docs/articles/workflow.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+REGCLIENT=${TOOLSDIR}/bin/regctl
+REGCLIENT_VERSION=v0.5.1
+curl -Lo ${REGCLIENT} https://github.com/regclient/regclient/releases/download/${REGCLIENT_VERSION}/regctl-linux-amd64
+chmod +x ${REGCLIENT}
+
+COSIGN=${TOOLSDIR}/bin/cosign
+COSIGN_VERSION=2.1.1
+curl -Lo ${COSIGN} https://github.com/sigstore/cosign/releases/download/v${COSIGN_VERSION}/cosign-linux-amd64 
+chmod +x ${COSIGN}
+
```
Comment (rchincha) [inline docs/articles/workflow.md:28]: ## Setup a OCI Registry With Referrers Support (`zot`)

File: docs/articles/workflow.md [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+function zot_teardown() {
+  killall zot
+}
+
+# call the function to start zot
+zot_setup
+{% endraw %}
+```
+
+
```
Comment (rchincha) [inline docs/articles/workflow.md:102]: ## Copy an image to the OCI Registry

File: docs/articles/workflow.md [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+  val: artifact
+EOF
+${REGCLIENT} artifact put --artifact-type application/yaml -f ${TEST_TMPDIR}/artifact.yaml --subject ${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+{% endraw %}
+```
+
+> :pencil2: If no subject is specified in the command, the artifact is considered independent and not associated with any existing image.
+
+
+### Step 5: Display the artifact tree
```
Comment (rchincha) [inline docs/articles/workflow.md:131]:
```
+ $ regctl artifact tree localhost:8080/busybox:latest
Ref: localhost:8080/busybox:latest
Digest: sha256:9172c5f692f2c65e4f773448503b21dba2de6454bd159905c4bf6d83176e4ea3
Referrers:
  - sha256:9c0655368b10ca4b2ffe39e4dd261fb89df25a46ae92d6eb4e6e1792a451883e: application/yaml
```

File: docs/articles/workflow.md [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+popd
+# sign the image
+COSIGN_PASSWORD= COSIGN_OCI_EXPERIMENTAL=1 COSIGN_EXPERIMENTAL=1 ${COSIGN} sign -y --key ${TEST_TMPDIR}/cosign.key --registry-referrers-mode=oci-1-1 ${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+# sign the artifact referring to the image
+COSIGN_PASSWORD= COSIGN_OCI_EXPERIMENTAL=1 COSIGN_EXPERIMENTAL=1 ${COSIGN} sign -y --key ${TEST_TMPDIR}/cosign.key --registry-referrers-mode=oci-1-1 ${ZOT_HOST}:${ZOT_PORT}/busybox@${REF0}
+{% endraw %}
+```
+
+
+### Step 7: Display the artifact tree
```
Comment (rchincha) [inline docs/articles/workflow.md:161]:
```+ $ regctl artifact tree localhost:8080/busybox:latest
Ref: localhost:8080/busybox:latest
Digest: sha256:9172c5f692f2c65e4f773448503b21dba2de6454bd159905c4bf6d83176e4ea3
Referrers:
  - sha256:9c0655368b10ca4b2ffe39e4dd261fb89df25a46ae92d6eb4e6e1792a451883e: application/yaml
    Referrers:
      - sha256:06792b209137486442a2b804b2225c0014e3e238d363cdbea088bbd73207fb34: application/vnd.dev.cosign.artifact.sig.v1+json
  - sha256:995b6a78bf04a7a9676dac76b4598ccb645c17e30b02f294de9fdfa2f28eb7b2: application/vnd.dev.cosign.artifact.sig.v1+json
```

Comment (rchincha) [reply docs/articles/workflow.md:161]: original image: localhost:8080/busybox:latest
Two direct refererrers:
artifact.yaml sha256:9c0655368b10ca4b2ffe39e4dd261fb89df25a46ae92d6eb4e6e1792a451883e: application/yaml
image signature sha256:995b6a78bf04a7a9676dac76b4598ccb645c17e30b02f294de9fdfa2f28eb7b2: application/vnd.dev.cosign.artifact.sig.v1+json

Second-level referrer for the artifact.yaml
- sha256:06792b209137486442a2b804b2225c0014e3e238d363cdbea088bbd73207fb34: application/vnd.dev.cosign.artifact.sig.v1+json

File: mkdocs.yml [hunk 9]
```diff
@@ -115,6 +115,7 @@ nav:
     - CI/CD Pipeline: articles/building-ci-cd-pipeline.md
     - Authentication and Authorization: articles/authn-authz.md
     - Verifying Image Signatures: articles/verifying-signatures.md
+    - Software Provenance Workflow Using OCI Artifacts: articles/workflow.md
```
Comment (rchincha) [inline mkdocs.yml:118]: Add this workflow.md article "link" in What's New (OCI artifacts support) - "See this article to do an end-to-end OCI artifacts workflow"

Comment (mbshields) [reply mkdocs.yml:118]: Done

File: docs/articles/workflow.md (hunk 4 above)
Comment (mbshields) [reply docs/articles/workflow.md]: Done

File: docs/articles/workflow.md (hunk 5 above)
Comment (mbshields) [reply docs/articles/workflow.md]: Done

File: docs/articles/workflow.md (hunk 6 above)
Comment (mbshields) [reply docs/articles/workflow.md:98]: Done

File: docs/articles/workflow.md [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+This step copies a busybox container image into the zot registry.
+
+```shell
+{% raw %}
+skopeo copy --format=oci --dest-tls-verify=false docker://busybox:latest docker://${ZOT_HOST}:${ZOT_PORT}/busybox:latest
+{% endraw %}
+```
+
+
+### Step 4: Copy an artifact to the zot registry
```
[the remark this answers is not in the document]
Comment (mbshields) [reply docs/articles/workflow.md]: Done

File: docs/articles/workflow.md (hunk 7 above)
Comment (mbshields) [reply docs/articles/workflow.md:127]: Command output added

File: docs/articles/workflow.md (hunk 8 above)
Comment (mbshields) [reply docs/articles/workflow.md:168]: Command output and explanation added

**Comment by rchincha** [comment]
> Questions: What is the purpose of creating and truncating REF0?
> 
> How is REF0 used in the cosign commands in Step 6?
> 
> Should we show an example of the artifact tree?

ref0 is for the artifact that was uploaded referring to the first container image.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**